tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -88,9 +93,158 @@ def contains_encrypted_indicator(line, encryption_field):
+                }
+            }
+        },
+        # TODO: tribal tanf
+    }
+
+    if query == "text":
+        for prog_name, prog_dict in schema_options.items():
+            for sect, val in prog_dict.items():
+                if val['section'] == section:
+                    return {'program_type': prog_name, 'section': sect}
+        raise ValueError("Model not found in schema_defs")
+    elif query == "section":
+        return schema_options.get(program, {}).get(section, None)[query]
+    elif query == "models":
+        links = schema_options.get(program, {}).get(section, None)
+
+        # if query is not chosen or wrong input, return all options
+        # query = 'models', model = 'T1'
+        models = links.get(query, links)
+
+        if model_name is None:
+            return models
+        elif model_name not in models.keys():
+            logger.debug(f"Model {model_name} not found in schema_defs")
+            return []  # intentionally trigger the error_msg for unknown record type
+        else:
+            return models.get(model_name, models)
+
+
+'''
+text -> section YES
+text -> models{} YES
+text -> model YES
+datafile -> model
+    ^ section -> program -> model
+datafile -> text
+model -> text YES
+section -> text
+
+text**: input string from the header/file
+'''
+
+def get_program_models(str_prog, str_section):
+    """Return the models dict for a given program and section."""
+    return get_schema_options(program=str_prog, section=str_section, query='models')
+
+def get_program_model(str_prog, str_section, str_model):
+    """Return singular model for a given program, section, and name."""
+    return get_schema_options(program=str_prog, section=str_section, query='models', model_name=str_model)
+
+def get_section_reference(str_prog, str_section):
+    """Return the named section reference for a given program and section."""
+    return get_schema_options(program=str_prog, section=str_section, query='section')
+
+def get_text_from_df(df):
+    """Return the short-hand text for program, section for a given datafile."""
+    return get_schema_options("", section=df.section, query='text')
+
+def get_prog_from_section(str_section):
+    """Return the program type for a given section."""
+    # e.g., 'SSP Closed Case Data'
+    if str_section.startswith('SSP'):
+        return 'SSP'
+    elif str_section.startswith('Tribal'):
+        return 'TAN'  # problematic, do we need to infer tribal entirely from tribe/fips code?
+    else:
+        return 'TAN'
+
+    # TODO: if given a datafile (section), we can reverse back to the program b/c the
+    # section string has "tribal/ssp" in it, then process of elimination we have tanf
+
+def get_schema(line, section, program_type):
+    """Return the appropriate schema for the line."""
+    line_type = line[0:2]
+    return get_schema_options(program_type, section, query='models', model_name=line_type)
+
+def fiscal_to_calendar(year, fiscal_quarter):
```

Comment:
@andrew-jameson i cant seem to get any aggregates with months other than oct, nov, dec. ive used several different quarters in the headers, the most recent one is below. 

![1613p4](https://github.com/raft-tech/TANF-app/assets/63075587/7f005f19-45d6-438c-a7e2-a70523b29261)

Reply:
nvm. im realizing its tied to frontend fiscal periods.